lib/assert.js

```diff
@@ -70,19 +70,11 @@ x.notSame = function (val, expected, msg) {
 
 x.throws = function (fn, err, msg) {
 	if (fn && fn.then) {
-		var isFailed = false;
-
 		return fn
-			.catch(function (err) {
-				isFailed = true;
-
-				return err;
+			.then(function () {
+				x.throws(function () {}, err, msg);
 			})
-			.then(function (fnErr) {
-				if (!isFailed) {
-					x.throws(function () {}, err, msg);
-				}
-
+			.catch(function (fnErr) {
```

Comment:
`x.throws(function () {}, err, msg);` throws an error, and is evaluated if `fn` isn't rejected.